### PR TITLE
Git fixes

### PR DESCRIPTION
**Pull request by wentasah**
Hi,

these are two fixes for problems that I experience with git submodules.

### CODE REVIEW

File: master/buildbot/steps/source/git.py [hunk 1]
```diff
@@ -534,14 +535,16 @@ def _syncSubmodule(self, _=None):
 
     def _updateSubmodule(self, _=None):
         if self.submodules:
-            return self._dovccmd(['submodule', 'update',
-                                  '--init', '--recursive', '--force'])
+            vccmd = ['submodule', 'update', '--init', '--recursive', '--force']
```
Comment (tardyp) [inline master/buildbot/steps/source/git.py]: looks like this one is not unit tested. We would apreciate if you add a unit test for the submodule use case.

Comment (wentasah) [inline master/buildbot/steps/source/git.py]: On Mon, Apr 27 2015, Pierre Tardy wrote:
>> @@ -534,14 +535,16 @@ def _syncSubmodule(self, _=None):
>>  
>>      def _updateSubmodule(self, _=None):
>>          if self.submodules:
>> -            return self._dovccmd(['submodule', 'update',
>> -                                  '--init', '--recursive', '--force'])
>> +            vccmd = ['submodule', 'update', '--init', '--recursive', '--force']
>
> looks like this one is not unit tested. We would apreciate if you add
> a unit test for the submodule use case.

It is - see https://github.com/wentasah/buildbot/commit/198f44f7241d3ec72a418bd9cca0b29aab3b488d

Or do you have in mind some other test?

-Michal

Comment (wentasah) [inline master/buildbot/steps/source/git.py]: Oh, you probably mean a test for pre-1.7.8 behavior. I'll add that.

**Comment by wentasah** [comment]
Is it better now? (https://github.com/wentasah/buildbot/commit/a310e30afab77fd6b04c9f728bf73ff7d8f4d847)

**Comment by tardyp** [comment]
:+1:

**Pull request merged**